render.go

```diff
@@ -3,47 +3,69 @@ package main
 import (
 	"bytes"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"os"
 	"strings"
 	"text/template"
 )
 
-var abbrs = []string{"id", "href", "url"}
+//** Render context **
 
-func (s *Schema) renderAsObject() string {
-	buf := bytes.NewBuffer([]byte{})
-	tmpl, err := template.New("struct.tmpl").Funcs(getFuncMap()).ParseFiles("templates/struct.tmpl")
-	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
-	}
-	err = tmpl.Execute(os.Stdout, s)
-	if err != nil {
-		fmt.Println(err)
-		os.Exit(2)
+type RenderContext struct {
+	templates       map[string]string
+	predefinedTypes map[string]string
+	abbrs           []string
+}
+
+func (c *RenderContext) array() (out []string) {
+	out = make([]string, 0)
+	for _, v := range c.templates {
+		out = append(out, v)
 	}
-	return buf.String()
+	return
 }
 
-func (s *Schema) renderAsArray() string {
-	buf := bytes.NewBuffer([]byte{})
-	tmpl, err := template.New("array.tmpl").Funcs(getFuncMap()).ParseFiles("templates/array.tmpl")
-	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
+func (c *RenderContext) find(key string) (name, path string, err error) {
+
+	path, ok := c.templates[key]
+	if !ok {
+		err = errors.New("invalid type")
+		return
 	}
-	err = tmpl.Execute(buf, s)
-	if err != nil {
-		fmt.Println(err)
-		os.Exit(2)
+
+	i := strings.LastIndex(path, "/")
+	if i < 0 {
+		name = path
+	} else {
+		name = path[i+1:]
 	}
 
-	return buf.String()
+	return
+}
+
+var renderContext = &RenderContext{
+	templates: map[string]string{
```

Comment:
we need to compile templates into binary. if we run go install in rott dir and then try to run oasgo somewhere else we will fail, because our binday will try to find templates dir